EasyFarm/States/SummonTrustsState.cs

```diff
@@ -0,0 +1,141 @@
+using System.Linq;
+using EasyFarm.Classes;
+using System.Collections.Generic;
+
+namespace EasyFarm.States {
+    public class SummonTrustsState : CombatState {
+        public Executor Executor { get; set; }
+
+        public SummonTrustsState(IMemoryAPI fface) : base(fface) {
+            Executor = new Executor(fface);
+        }
+
+        private bool PartyHasSpace() {
+            var slots = 0;
+            for(int i = 1; i < 6; i++) {
+                if (!fface.PartyMember[(byte)i].UnitPresent) slots++;
+            }
+            return slots > 0;
+        }
+
+        private IPartyMemberTools FindPartyMember(BattleAbility trust) {
+            for (int i = 1; i < 6; i++) {
+                var p = fface.PartyMember[(byte)i];
+                var comp = trust.Name;
+                if (comp.Contains("(UC)") || comp.Contains("II")) {
+                    comp = comp.Replace(" (UC)", "");
+                    comp = comp.Replace(" II", "");
+                }
+                if (p.UnitPresent && p.Name == comp) {
+                    return p;
+                }
+            }
+            return null;
+        }
+
+        private bool TrustInParty(BattleAbility trust) {
+            var t = FindPartyMember(trust);
+            return (t != null);
+        }
+
+        private bool TrustNeedsSummoning(BattleAbility trust) {
+            if (TrustInParty(trust) && TrustNeedsDismissal(trust)) {
+                ReleaseTrust(trust);
+            }
+            return (!TrustInParty(trust) && PartyHasSpace());
+        }
+
+        private bool TrustNeedsDismissal(BattleAbility trust) {
+            var t = FindPartyMember(trust);
+            if (t == null) return false;
+
+            // If the trust is set to be resummonable, respect the MP.
+            if (trust.ResummonOnLowMP)
+            {
+                if (t.MPPCurrent <= trust.ResummonMPHigh && t.MPPCurrent >= trust.ResummonMPLow)
+                {
+                    return true;
+                }
+            }
+
+            // If the trust is set to be resummonable, respect the HP 
+            if (trust.ResummonOnLowHP)
+            {
+                if (t.HPPCurrent <= trust.ResummonHPHigh && t.HPPCurrent >= trust.ResummonHPLow)
+                {
+                    return true;
+                }
+            }
+        
+            return false;
+        }
+
+        private void ReleaseTrust(BattleAbility trust) {
+            var comp = trust.Name;
+            if (comp.Contains("(UC)") || comp.Contains("II"))
+            {
+                comp = comp.Replace(" (UC)", "");
+                comp = comp.Replace(" II", "");
+            }
+
+            var command = string.Format("/refa {0}", comp);
+            fface.Windower.SendString(command);
+        }
+
+        public override bool Check() {
+            if (new RestState(fface).Check()) return false;
+            if (
+                fface.Player.Status.Equals(Status.CatchMonster) ||
```

Comment:
We can use a blacklist here `if(!fface.Player.Status.Equals(Status.Standing))` to simplify the check. 

The `Helpers.ToStatus` method in the MemoryAPI project has a list of which statuses are currently in use. The other statuses are returned as `Status.Unknown`. 

[EasyFarm/MemoryAPI/Helpers.cs](https://github.com/EasyFarm/EasyFarm/blob/4962b76cccb168c46ddad2e757417aebb8273a9f/MemoryAPI/Helpers.cs#L24)